sdk/go/common/resource/plugin/provider_server.go

```diff
@@ -144,7 +144,12 @@ func (p *providerServer) Handshake(
 		return nil, err
 	}
 
-	return &pulumirpc.ProviderHandshakeResponse{}, nil
+	// As in Configure, we always accept secrets, resources, and outputs.
+	return &pulumirpc.ProviderHandshakeResponse{
+		AcceptSecrets:   true,
```

Comment:
How do we opt into the other features here?

Reply:
Good shout. I've fixed this so that autonaming is respected; the others are (I think) always true as in `Configure` since `providerServer` can always shim them I believe.